src/App.tsx

```diff
@@ -1,14 +1,67 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [status, setStatus] = useState('all');
+  const [query, setQuery] = useState('');
+  let visibleTodos = todosFromServer;
+  const todoToModal = visibleTodos.filter(todo => todo.id === selectedTodoId);
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      const todos = await getTodos();
+
+      setTodosFromServer(todos);
+    };
+
+    getTodosFromServer()
+      .catch(Error);
+  }, []);
+
+  const selectTodo = (id: number) => {
+    if (selectedTodoId !== id) {
+      setSelectedTodoId(id);
+    } else {
+      setSelectedTodoId(0);
+    }
+  };
+
+  switch (status) {
+    case 'all':
+      visibleTodos = todosFromServer;
+      break;
+
+    case 'active':
+      visibleTodos = visibleTodos.filter(todo => !todo.completed);
+      break;
+
+    case 'completed':
+      visibleTodos = visibleTodos.filter(todo => todo.completed);
+      break;
+    default:
+      break;
+  }
```

Comment:
As the second option, here you can use a switch case inside of the filter method, where in the default case you need just to return true like you choose 'all'. Just as one more option)
```suggestion
  visibleTodos = visibleTodos.filter(todo => {
     switch (status) {
         case 'active':
             return !todo.completed;
         case 'completed':
            return todo.completed;
         default:
            return true;
     }
  });
```